cmd/controller-manager/app/controller-manager.go

```diff
@@ -301,9 +301,23 @@ func updateFederationConfig(config *rest.Config, fedConfig *corev1a1.FederationC
 func setOptionsByFederationConfig(opts *options.Options) {
 	fedConfig := getFederationConfig(opts)
 	if fedConfig == nil {
+		// Computing the right value of the namespace based on how
+		// federation is deployed.
+		if opts.Scope == apiextv1b1.NamespaceScoped {
+			opts.Config.TargetNamespace = opts.Config.FederationNamespace
+			glog.Infof("Federation will be limited to the %q namespace", opts.Config.FederationNamespace)
+		} else {
+			opts.Config.TargetNamespace = metav1.NamespaceAll
+			glog.Info("Federation will target all namespaces")
+		}
 		// FederationConfig could not be sourced from --federation-config or from the API.
 		// create a new `FederationConfig` with default values.
-		fedConfig = &corev1a1.FederationConfig{}
+		fedConfig = &corev1a1.FederationConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      util.FederationConfigName,
+				Namespace: opts.Config.TargetNamespace,
```

Comment:
The correct value for the namespace of the federation config is `opts.Config.FederationNamespace`.  This value is set via the `--federation-namespace` [argument](https://github.com/kubernetes-sigs/federation-v2/blob/master/charts/federation-v2/charts/controllermanager/templates/deployment.yaml#L26) which is intended to be [set via the downward api](https://github.com/kubernetes-sigs/federation-v2/blob/master/charts/federation-v2/charts/controllermanager/templates/deployment.yaml#L37) to the namespace the controller manager is running in.

Reply:
Thanks for this detailed explanation.